superset-frontend/src/components/Select/styles.tsx

```diff
@@ -245,18 +245,18 @@ const { ClearIndicator, DropdownIndicator, Option } = defaultComponents;
 
 export const DEFAULT_COMPONENTS: SelectComponentsConfig<any> = {
   Option: ({ children, innerProps, data, ...props }) => (
-    <Option
-      {...props}
-      data={data}
-      innerProps={{
-        ...innerProps,
-        // `@types/react-select` didn't define `style` for `innerProps`
-        // @ts-ignore
-        style: data && data.style ? data.style : null,
-      }}
-    >
-      {children}
-    </Option>
+    <ClassNames>
```

Comment:
This is cool, and looks like it'll work... but I'm wondering if there's any advantage to this approach vs the `` const Styles = styled.div`...` `` approach

Reply:
Hmm, I think this approach might be a bit cleaner. If we used `styled.div` we'd also need to use `react-select`'s API to get default styles and merge them with our `data.styles`, so in my opinion it's not worth it. WDYT?